src/confopt/profile/base.py

```diff
@@ -231,18 +402,57 @@ def _set_perturb(
         self._initialize_perturbation_config()
 
     def _set_partial_connector(self, is_partial_connection: bool = False) -> None:
+        """Set the value is_partial_connection along the defualt configuration for
+        the partial connector.
+
+        Args:
+            is_partial_connection (bool): Flag to enable partial connections in \
+                the supernet. Defaults to False.
+
+        Returns:
+            None
+        """
         self.is_partial_connection = is_partial_connection
         self._initialize_partial_connector_config()
 
     def _set_dropout(self, dropout: float | None = None) -> None:
+        """Set the value of dropout operation for the architecture parameters \
+            along with the default configurations for dropout.
+
+        Args:
+            dropout (float | None): Dropout operation rate for architectural \
+                parameters. Defaults to None.
+
+        Returns:
+            None
+        """
         self.dropout = dropout
         self._initialize_dropout_config()
 
     def _set_regularization(self, is_regularization_enabled: bool = False) -> None:
+        """Set the value of is_regularization_enabled along with the default \
+            configuration for regularization.
+
+        Args:
+            is_regularization_enabled (bool): Flag to enable regularization \
+                during training. Defaults to False.
+
+        Returns:
+            None
+        """
         self.is_regularization_enabled = is_regularization_enabled
         self._initialize_regularization_config()
 
     def get_config(self) -> dict:
+        """This method returns a dictionary containing the configurations for \
```

Comment:
Improve phrasing to indicate that the config is essentially a dictionary representation of the Profile class